core/services/pipeline/models.go

```diff
@@ -158,18 +176,24 @@ func (tr TaskRun) Result() Result {
 	return result
 }
 
+func (tr *TaskRun) IsPending() bool {
+	return !tr.FinishedAt.Valid && tr.Output.Empty() && tr.Error.IsZero()
+}
+
 // RunStatus represents the status of a run
-type RunStatus int
+type RunStatus string
 
 const (
 	// RunStatusUnknown is the when the run status cannot be determined.
-	RunStatusUnknown RunStatus = iota
-	// RunStatusInProgress is used for when a run is actively being executed.
-	RunStatusInProgress
+	RunStatusUnknown RunStatus = "unknown"
```

Comment:
I'm still not sure this "unknown" status is necessary

Reply:
I got rid of it in the DB side, but this is still used in some of the flux monitor code.